app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Api::V1::UsersController < ApplicationController
+  def index
+    json_create(User.all)
+  end
+
+  def show
+    json_create(User.find(params[:id]))
+  end
+
+  def create
+    if User.find_by(email: params[:email])
+      json_create(User.find_by(email: params[:email]))
+    else
+      user = User.new(user_params)
+      if user.save
+        json_create(user)
+      else
+        render json: { data: { errors: user.errors.full_messages.to_sentence, status: 400 } },
+               status: :bad_request
+      end
+    end
+  end
```

Comment:
Did this assuming the frontend will send us a post request for a login. If the user exists we will send back the user if not we will make the new user and send them that new object.  missing name or email will result in error